### PR TITLE
Fix/markdown links

### PR DESCRIPTION
**Pull request by MaddyDev**
Still not sure if this is the right fix. But we don't own any components of markdown files preview and use vscode webview editor to display markdown files from where the relative links path is being passed. 

The issue that I see is, path (relative path) that's being passed from the markdown file is supposed to be file URI after the vscode merge but it's still being passed just as a string path which when converted is losing some essential data and is not a valid url. 
Added an extra check in the OpenDocumentLinkCommand of markdown extension, to check if path is string and resource is file and handle it as file instead of url.

This PR addresses #11657

### CODE REVIEW

File: extensions/markdown-language-features/src/commands/openDocumentLink.ts [hunk 1]
```diff
@@ -58,7 +59,7 @@ export class OpenDocumentLinkCommand implements Command {
 
 	public static async execute(engine: MarkdownEngine, args: OpenDocumentLinkArgs) {
 		const fromResource = vscode.Uri.parse('').with(args.fromResource);
-		const targetResource = vscode.Uri.parse('').with(args.path);
+		const targetResource = fromResource.scheme === 'file' && isString(args.path) ? vscode.Uri.file(args.path) : vscode.Uri.parse('').with(args.fromResource);
```
Comment (Charles-Gagnon) [inline extensions/markdown-language-features/src/commands/openDocumentLink.ts:62]: The second part should be `vscode.Uri.parse('').with(args.path)` - keeping it the same as it is currently.

**Comment by Charles-Gagnon** [comment]
Here's the specific change that broke this - the args were updated to specify path as an object (URI?) but the param being passed in was left as a string https://github.com/microsoft/vscode/commit/fdf23dc5f95f4a17b246dbebc9af14dab5043334#diff-ac54e693c637dc86bf104bdc419aa7dc

File: extensions/markdown-language-features/src/commands/openDocumentLink.ts (hunk 1 above)
Comment (MaddyDev) [reply extensions/markdown-language-features/src/commands/openDocumentLink.ts:62]: oops missed that, thanks!

**Comment by MaddyDev** [comment]
> Here's the specific change that broke this - the args were updated to specify path as an object (URI?) but the param being passed in was left as a string [microsoft/vscode@fdf23dc#diff-ac54e693c637dc86bf104bdc419aa7dc](https://github.com/microsoft/vscode/commit/fdf23dc5f95f4a17b246dbebc9af14dab5043334#diff-ac54e693c637dc86bf104bdc419aa7dc)

Yes that's exactly it. I'll create a bug for this in vscode. Thanks

**Comment by MaddyDev** [comment]
Created an [vscode issue](https://github.com/microsoft/vscode/issues/104267)

**Comment by Charles-Gagnon** [comment]
@MaddyDev FYI - don't forget the SQL CARBON EDIT for any VS Code code

**Pull request merged**